### PR TITLE
feat: add support for PHP 8

### LINKED ISSUE #3: PHP 8.0 support
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes

#### Summary

To be prepared for the december release of PHP 8.0, this repository has some additional TODOs to be tested against the new major version.

In order to make this repository compatible, one has to follow these steps:

- [ ] Modify `composer.json` to provide support for PHP 8.0 by adding the constraint `~8.0.0`
- [ ] Modify `composer.json` to drop support for PHP less than 7.3
- [ ] Modify `composer.json` to implement phpunit 9.3 which supports PHP 7.3+
- [ ] Modify `.travis.yml` to ignore platform requirements when installing composer dependencies (simply add `--ignore-platform-reqs` to `COMPOSER_ARGS` env variable)
- [ ] Modify `.travis.yml` to add PHP 8.0 to the matrix (*NOTE*: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [ ] Modify source code in case there are incompatibilities with PHP 8.0

### PR DESCRIPTION
**Pull request by owenvoke**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | no

### Description

This adds support for PHP 8. It looks like Travis doesn't seem to support PHP 8 yet though (I haven't used Travis for ages)?

Closes #3

### CODE REVIEW

**Comment by rieschl** [comment]
> This adds support for PHP 8. It looks like Travis doesn't seem to support PHP 8 yet though (I haven't used Travis for ages)?


Yes you still have to use `nightly`.

**Comment by owenvoke** [comment]
Thanks, I've now updated it to use Nightly.

**Comment by froschdesign** [comment]
@owenvoke 
Thanks for your contribution! 👍 
Please look at the related task list for support PHP 8: #3

**Comment by owenvoke** [comment]
I'm following that task list. 👍🏻  Currently on the last one, but it looks like the only issue in the tests is the `servicemanager` which needs to be updated to a minimum of `3.2.1` as far as I can tell. Is that ok to do? Or do we still need to support `2.x`, which doesn't seem to have the changes.

**Comment by froschdesign** [comment]
> I'm following that task list.

You have missed an important task:

> Modify `composer.json` to drop support for PHP less than 7.3

Which also means: remove the old version from the Travis configuration. 😉

File: composer.json [hunk 1]
```diff
@@ -25,15 +25,15 @@
         }
     },
     "require": {
-        "php": "^5.6 || ^7.0",
+        "php": "^7.3 || ^8.0",
```
Comment (rieschl) [inline composer.json]: Please use `~8.0.0` as mentioned in the issue. That way the package supports PHP up to `8.0.*`.

**Review by rieschl — CHANGES_REQUESTED** [review]

**Comment by owenvoke** [comment]
Ah, makes sense. 🤦🏻 I've removed the old versions, and updated to a minum of Service Manager v3.4, which added PHP8 support (and dropped <7.3). And also now removed the old versions from Travis. 😬 Apologies for forgetting this.

**Comment by boesing** [comment]
Thanks @owenvoke

**Pull request merged**